src/helpers/generator.ts

```diff
@@ -22,52 +22,25 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-import type { ActionInContext } from "@elastic/synthetics";
-import type { Action, Step, Steps } from "../common/types";
+import type { Action, ActionInContext, Step, Steps } from "@elastic/synthetics";
 
-export function generateIR(actionContexts: Step) {
-  const result = [];
-  let steps = [];
-  let previousContext = null;
-  let newStep = false;
-  for (const actionContext of actionContexts) {
-    const { action, pageAlias, title } = actionContext;
-    if (action.name === "openPage") {
-      continue;
-    } else if (action.name === "closePage" && pageAlias === "page") {
-      continue;
+export function generateIR(steps: Steps): Steps {
+  const result: Steps = [];
+  const actions: ActionInContext[] = [];
+  for (const step of steps) {
+    for (const actionContext of step.actions) {
+      const { action, title } = actionContext;
+      // Add title to all actionContexts
+      actions.push(
+        title ? actionContext : { ...actionContext, title: actionTitle(action) }
```

Comment:
Can you expose the actionTitle from synthetics on the step generator PR? I saw we added `assert` as well.

Reply:
++ will do that now

Reply:
https://github.com/elastic/synthetics/pull/457/commits/113c5d794aea2bf26b829df3f5115dafef9b7b09